src/contexts/events.tsx

```diff
@@ -143,12 +153,28 @@ const useEvents = () => {
     trackedEventTypes.current = new Set();
   }, []);
 
+  useEffect(() => {
+    if (!chainId) return;
+
+    if (previousChainId.current === undefined) {
+      previousChainId.current = chainId;
+      // This means we are in the first render, so we don't want to track the event
+      return;
+    }
+
+    trackEvent({
+      event: 'network_change',
+      from_network: previousChainId.current || chainId,
```

Comment:
We already check if `previousChainId.current` is undefined, we don't have to `|| chainId`